cdragontoolbox/wad.py

```diff
@@ -277,15 +277,20 @@ def set_unknown_paths(self, path):
                     wadfile.path = f"{path}/{wadfile.path_hash:016x}"
 
     def sanitize_paths(self):
-        """Truncate files whose basename has a length of at least 255"""
+        """Sanitize paths for extract purposes; for example truncating files whose basename has a length of at least 255"""
 
         for wadfile in self.files:
             if wadfile.path:
                 path, filename = os.path.split(wadfile.path)
+                basename, ext = os.path.splitext(filename)
+                if wadfile.ext and not ext:
+                    # extension was guessed, but the resolved path has no extension
+                    # in this case, append the guessed extension with custom suffix
+                    wadfile.path = f"{wadfile.path}.cdtb.{wadfile.ext}"
+
                 if len(filename) < 255:
```

Comment:
For clarity I would change this to
```py
if len(filename) >= 255:
    wadfile.path = os.path.join(path, f"{basename[:255-17-len(ext)]}.{wadfile.path_hash:016x}{ext}")
```

This makes it clearer that there are two different change being checked for and applied.